app/src/main/res/values/strings.xml

```diff
@@ -232,13 +232,13 @@
     <string name="unsupport_magisk_msg">This version of the app does not support Magisk version lower than %1$s.\n\nThe app will behave as if no Magisk is installed, please upgrade Magisk as soon as possible.</string>
     <string name="unsupport_general_title">Abnormal State</string>
     <string name="unsupport_system_app_msg">Running this app as a system app is not supported. Please revert the app to a user app.</string>
-    <string name="unsupport_other_su_msg">An \"su\" command that does not belong to Magisk is detected. Please remove the other unsupported su.</string>
+    <string name="unsupport_other_su_msg">Please remove the app that just tried to handle a \"su\" command and handle all requests through Magisk instead.</string>
     <string name="unsupport_external_storage_msg">Magisk is installed to external storage. Please move the app to internal storage.</string>
-    <string name="unsupport_nonroot_stub_msg">The app cannot continue to work in the hidden state as root was lost. Please restore it back to the original APK.</string>
```

Comment:
"state" is redundant here.

Reply:
Unless you translate it, then all balls are in the air.
Keep it simple, and keep it short. More text and concepts equates to higher cognitive load, more stress, and less confidence as a result.
This is a win.

Reply:
The original "continue to work" was better. This falls under you changing too much again

Reply:
It's a better way of explaining it.

Reply:
It does the opposite of explaining.
Either it stops working, _and_ it is in the hidden state
or
It stops being able to be in the hidden state.

Moreover "The app cannot continue to work in the hidden state as root" is not meaningful until "was lost" is added.

Reply:
> Either it stops working, and it is in the hidden state
> or
> It stops being able to be in the hidden state.

Those two don't exclude each other. The app is in the hidden state, then the user loses root for some reason, and when the user enters the app receives that error message and is given the ability to "un-hide" the app (aka restoring it to the original APK).

> Moreover "The app cannot continue to work in the hidden state as root" is not meaningful until "was lost" is added.

"Since root permissions were lost, the app cannot continue to work in the hidden state anymore" sounds better?

Reply:
This is a case of "you can write it any of these ways" because English. 😛

As means because here, if that is tricky in translation then let's just be clear:
```
The hidden app cannot continue to work because root was lost. Please restore it to the original APK.
```

Removed some redundant language in there too to make you happier. 😉

Reply:
Sentence one I can tolerate, but it seems to suggest there is one hidden Magisk, and one non-hidden Magisk.
That is if the user understands "the hidden app" as Magisk.
If two different Magisk apps is technically true, it is confusing to bring up like that.

The second sentence is off somehow, and I think it is because both an app and root can be restored. "… to the original APK" keeps entertaining this duality.
How about:
"Magisk has lost the root access it used to stay hidden. Please reestablish it."?
or something like
""Magisk must be uninstalled because it has lost the root access it used to stay hidden. Please reestablish root in a fresh version of the app."? (if that is how it works)

Reply:
Those are all crazy wordy and don't seem to capture the necessary information. Have you ever run hidden Magisk after root is lost (e.g. you flash stock boot image back)? Maybe you should try so you understand the situation better.

Reply:
The second option is the shortest yet.
I'd only be using my own nb_NO translation in so doing, which is as vague as the original because I wasn't 100% sure at the time.
In German and Italian it says specifically to recover the app to its original state. It is not translated in Swedish or Spanish. There is no Danish translation.

When you say "capture necessary information" that does sound bad, but I don't know why.
Here is my problem with the proposals:
>The hidden app cannot continue to work because root was lost. Please restore it to the original APK.
`The hidden app`
What is the hidden app (unanswered)
`cannot continue to work`
how?
`because root was lost.`
So by regaining root one would think "the hidden app" would continue to work.
It doesn't say whether this is general root access, or root access granted to "the hidden app"
`Please restore it to the original APK.`
"It" is unexplained here. It could either be
~root access to the original APK (presumably of "the hidden app")", or
~"the hidden app" restored into its original APK form."

vs

"Grant Magisk the root access it needed to stay hidden to hide it again, or unhide it now."

`Grant Magisk`
Direct action, we know what app it is.
`the root access`
We know the quality of the action
`it needed to stay hidden`
It was hidden, now it isn't. Root was lost. Thinking it can then be rehidden.
`to hide it again`,
Causality. This is at least a precursor to getting there. Further action may be required.
`or unhide it now.`
option b, which is the lesser of the two.
The user knows what is going on, and what the differing option entails.
All risk of the system discovering Magisk is undertaken at the risk of the user doing it, but it is not disclosed.
There is clear preference for re-rooting and keeping Magisk hidden, which has to cause fewer errors.

Reply:
You can't grant root to it though, root was lost. Again you don't seem to understand the problem the string applies to.

Reply:
Not if the user didn't install Magisk, or forgot, or doesn't for whatever reason not make the connection.
Or in other words, the understanding a central concept is only available given the recollected experience of installing the app, and then solving for a lot of unknown factors, in the form of a riddle.
It relies heavily on technical proficiency.
Other than there not being an option to re-root, why are those two sentences the best possible solution for mitigating my concerns?

The relationship between "Magisk", "hidden" and "root" can be perfectly clear without relying on context.

If there is no other option than to unhide the app, why isn't it:
"Magisk will be un-hidden (and restored to its original state) because root was lost."
"APK" and "restore" isn't there, but is that something the user has to know about? If so, why?
Do you have concerns with it?

Reply:
That's ridiculous and frankly it feels like you're being intentionally obtuse. Nobody is opening the hidden Magisk app and not knowing what it is. We can't rewrite everything in the app for some imaginary idiot you envision who has no idea how anything on their device has occurred. 😕🤨

Reply:
But "stay hidden" somehow is worse than "the hidden state"?

Magisk has fewer users than it deserves, because it is a great tool.
No matter how much better it gets, these types of considerations is where it either looks scary or not.

I am not the most technical user, which makes me the bulk of the current audience.
I think readability greatly enhances my enjoyment.
Think of the time it takes to decipher the meaning of something as comparable latency on the back-end.
Just because it happens at a different interfaces doesn't make it OK.

All the translations I looked at directly translate difficult passages, and that is not good.
Sure, only the users one doesn't wish for have a less than stellar command of English, but at the best of times that is you and me both.

Reply:
I don't think we're going to get any further here. My suggestions are about as good as they're going to get at the moment, from both a developer and user standpoint. I think you need to go with mine here so we can move forward with the rest of it.

Reply:
That explanation I understand. I went with your version and clarified my concerns with it.